conslogging/conslogging.go

```diff
@@ -266,6 +272,72 @@ func (cl ConsoleLogger) PrefixColor() *color.Color {
 	return cl.color(c)
 }
 
+// Prints a GitHub Actions summary message to GITHUB_STEP_SUMMARY
+func (cl *ConsoleLogger) PrintGHASummary(message string) {
+	if !cl.isGitHubActions {
+		return
+	}
+
+	path := os.Getenv("GITHUB_STEP_SUMMARY")
+	if path == "" {
+		return
+	}
+	file, err := os.OpenFile(path, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return
+	}
+	defer file.Close()
+	_, _ = file.WriteString(message + "\n")
+}
+
+// PrintGHAError constructs a GitHub Actions error message.
+// The `file`, `line`, and `col` parameters are optional.
+func (cl *ConsoleLogger) PrintGHAError(message string, details ...string) {
+	cl.mu.Lock()
+	defer cl.mu.Unlock()
+
+	file := ""
+	line := ""
+	col := ""
+
+	if len(details) >= 3 {
```

Comment:
having a different code path based on the length here is a bit hard to follow; a better go idiom is to define a functional option, see https://golang.cafe/blog/golang-functional-options-pattern.html

This would allow you to call it with:

```
PrintGHAError( "my error message", WithGHASourceLocation("my-file", 10, 20))
```

or

```
PrintGHAError( "my error message")
```

which means the code could look like:

```
type GHAError struct {
  message string
  file string
  line int
  col int
}

type GHAErrorOpt func(*GHAError)

func WithGHASourceLocation(file string, line, col int) GHAErrorOpt {
  return func(cfg *GHAError) {
    cfg.file = file
    cfg.line = line
    cfg.col = col
  }
}

func (cl *ConsoleLogger) PrintGHAError(message string, fns ...GHAErrorOpt) {
    cfg := GHAError{
        message: message,
    }
    for _, fn := range fns {
      fn(&cfg)
    }

  if cfg.file != "" {
    cl.printGithubActionsControl(errorCommand, "file=%s,line=%s,col=%s,title=Error::%s", file, line, col, message)
  } else {
    cl.printGithubActionsControl(errorCommand, "title=Error::%s", message)
  }
}
```

Reply:
done